bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -516,6 +524,33 @@ impl Connection {
         }
     }
 
+    /// Check if client auth was used for a connection.
+    ///
+    /// This is only relevant if [`ClientAuthType::Optional] was used.
+    pub fn client_cert_used(&self) -> bool {
+        unsafe { s2n_connection_client_cert_used(self.connection.as_ptr()) == 1 }
+    }
+
+    /// Retrieves the raw bytes of the client cert chain received from the peer, if present.
+    pub fn client_cert_chain_bytes(&self) -> Result<Option<&[u8]>, Error> {
```

Comment:
I thought we usually don't hand out pointers to applications...

Reply:
Apparently we decided to do it anyway for certificates. We do the same for the newer certificate APIs too: https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L2336-L2338

Reply:
I could make the bindings allocate memory, or I could add lifetimes to the method. I'm leaning towards lifetimes.

Reply:
Well the nice thing is we get lifetimes for free here 🙂. I was just surprised that we had apis that handed out pointers